### ISSUE TITLE
The maven plugin fails with multiple exceptions "java.lang.ArrayIndexOutOfBoundsException: NN"

### ISSUE THREAD

**Issue by NikitaKriukov** [spotbugs/spotbugs#2269]
I faced it accidentally in our project. I didn't manage to find the reason, because the exception doesn't say anything valuable. It only contains multiple ArrayIndexOutOfBoundsException exceptions. I prepared a small project, which causes the series of exceptions. It should be helpful to reproduce the issue. 

The project has only 1 small Java class. It has minimum number of code. Some fragments of code can seem useless, but any changes can lead to the exception disappearance.

**Comment by NikitaKriukov** [comment]
Here is the Java class: 

`package aaa.bbb.ccc;

import lombok.NonNull;

import java.util.function.Consumer;

// This class causes SpotBugs Maven plugin exception
public class SpotBugsTest {

    public void func(Consumer<Consumer<State>> consumer, Consumer<Runnable> consumer2) {
        consumer.accept(
            (State state) -> {
                consumer2.accept(() -> {
                    System.out.println("DUMMY");
                    @NonNull String subState = state.getSubState();
                    System.out.println("DUMMY");
                });
            }
        );
    }

    private static class State {
        String getSubState() {
            return null;
        }
    }

    // Don't delete it!
    static class Dummy implements Comparable<Dummy> {
        private String str1;

        @Override
        public int compareTo(Dummy other) {
            return 0;
        }
    }

}`

**Comment by NikitaKriukov** [comment]
Here is Maven pom.xml

<?xml version="1.0" encoding="utf-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>spot-bugs-test</groupId>
    <artifactId>spot-bugs-test</artifactId>
    <version>1.0</version>
    <packaging>jar</packaging>
    <name>spot-bugs-test</name>
    <properties>
        <maven.compiler.source>1.8</maven.compiler.source>
        <maven.compiler.target>1.8</maven.compiler.target>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    </properties>
    <dependencies>
        <dependency>
            <groupId>org.apache.commons</groupId>
            <artifactId>commons-text</artifactId>
            <version>1.10.0</version>
        </dependency>
        <dependency>
            <groupId>org.projectlombok</groupId>
            <artifactId>lombok</artifactId>
            <version>1.18.22</version>
        </dependency>
        <dependency>
            <groupId>com.google.guava</groupId>
            <artifactId>guava</artifactId>
            <version>31.1-jre</version>
        </dependency>
        <dependency>
            <groupId>org.apache.commons</groupId>
            <artifactId>commons-lang3</artifactId>
            <version>3.12.0</version>
        </dependency>
    </dependencies>
    <build>
        <plugins>
            <plugin>
                <groupId>com.github.spotbugs</groupId>
                <artifactId>spotbugs-maven-plugin</artifactId>
                <version>4.7.3.0</version>
                <configuration>
                    <effort>Max</effort>
                    <xmlOutput>true</xmlOutput>
                    <excludeFilterFile>build_config/findbugs-exclude.xml</excludeFilterFile>
                </configuration>
                <executions>
                    <execution>
                        <goals>
                            <goal>check</goal>
                        </goals>
                        <phase>verify</phase>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>
</project>
`
`

**Comment by NikitaKriukov** [comment]
Pay attention the example uses Lombok library. In particular the @NonNull annotation.

**Comment by Ceddaerrix** [comment]
I am experiencing the same behavior in my multiple-module project.

In the parent pom.xml, I have the plugin set as follow in the <build> section:
```
<plugin>
  <groupId>com.github.spotbugs</groupId>
  <artifactId>spotbugs-maven-plugin</artifactId>
  <version>4.7.3.0</version>
	<executions>
		<execution>
			<phase>verify</phase>
			<goals>
				<goal>spotbugs</goal>
			</goals>
		</execution>
	</executions>
	<configuration>
		<failsOnError>false</failsOnError>
		<relaxed>true</relaxed>
		<xmlOutput>true</xmlOutput>
		<effort>more</effort>
	</configuration>
</plugin>
```

then, the build console output looks like (trimmed):
```
...
[INFO] 
[INFO] --- spotbugs-maven-plugin:4.7.3.0:spotbugs (default) @ y-project-module6 ---
[INFO] Fork Value is true
     [java] The following errors occurred during analysis:
     [java]   Exception analyzing com.example.in.some.package.Class1 using detector edu.umd.cs.findbugs.detect.RuntimeExceptionCapture
     [java]     java.lang.ArrayIndexOutOfBoundsException: 5
     [java]       At edu.umd.cs.findbugs.BugCounts.addError(PackageStats.java:50)
     [java]       At edu.umd.cs.findbugs.PackageStats.addError(PackageStats.java:240)
     [java]       At edu.umd.cs.findbugs.ProjectStats.addBug(ProjectStats.java:272)
     [java]       At edu.umd.cs.findbugs.SortedBugCollection.add(SortedBugCollection.java:960)
     [java]       At edu.umd.cs.findbugs.SortedBugCollection.add(SortedBugCollection.java:227)
     [java]       At edu.umd.cs.findbugs.BugCollectionBugReporter.doReportBug(BugCollectionBugReporter.java:100)
     [java]       At edu.umd.cs.findbugs.AbstractBugReporter.reportBug(AbstractBugReporter.java:179)
     [java]       At edu.umd.cs.findbugs.DelegatingBugReporter.reportBug(DelegatingBugReporter.java:69)
     [java]       At edu.umd.cs.findbugs.FilterBugReporter.reportBug(FilterBugReporter.java:49)
     [java]       At edu.umd.cs.findbugs.BugAccumulator.reportBug(BugAccumulator.java:199)
     [java]       At edu.umd.cs.findbugs.BugAccumulator.reportAccumulatedBugs(BugAccumulator.java:180)
     [java]       At edu.umd.cs.findbugs.detect.RuntimeExceptionCapture.visitJavaClass(RuntimeExceptionCapture.java:118)
     [java]       At org.apache.bcel.classfile.JavaClass.accept(JavaClass.java:213)
     [java]       At edu.umd.cs.findbugs.BytecodeScanningDetector.visitClassContext(BytecodeScanningDetector.java:38)
     [java]       At edu.umd.cs.findbugs.DetectorToDetector2Adapter.visitClass(DetectorToDetector2Adapter.java:76)
     [java]       At edu.umd.cs.findbugs.FindBugs2.lambda$analyzeApplication$1(FindBugs2.java:1108)
     [java]       At java.util.concurrent.FutureTask.run(FutureTask.java:266)
     [java]       At edu.umd.cs.findbugs.CurrentThreadExecutorService.execute(CurrentThreadExecutorService.java:86)
     [java]       At java.util.concurrent.AbstractExecutorService.invokeAll(AbstractExecutorService.java:238)
     [java]       At edu.umd.cs.findbugs.FindBugs2.analyzeApplication(FindBugs2.java:1118)
     [java]       At edu.umd.cs.findbugs.FindBugs2.execute(FindBugs2.java:309)
     [java]       At edu.umd.cs.findbugs.FindBugs.runMain(FindBugs.java:395)
     [java]       At edu.umd.cs.findbugs.FindBugs2.main(FindBugs2.java:1231)
     [java]   Exception analyzing com.example.in.another.package.Class2 using detector edu.umd.cs.findbugs.detect.RuntimeExceptionCapture
     [java]     java.lang.ArrayIndexOutOfBoundsException: 5
     [java]       At edu.umd.cs.findbugs.BugCounts.addError(PackageStats.java:50)
     [java]       At edu.umd.cs.findbugs.PackageStats.addError(PackageStats.java:240)
     [java]       At edu.umd.cs.findbugs.ProjectStats.addBug(ProjectStats.java:272)
     [java]       At edu.umd.cs.findbugs.SortedBugCollection.add(SortedBugCollection.java:960)
     [java]       At edu.umd.cs.findbugs.SortedBugCollection.add(SortedBugCollection.java:227)
     [java]       At edu.umd.cs.findbugs.BugCollectionBugReporter.doReportBug(BugCollectionBugReporter.java:100)
     [java]       At edu.umd.cs.findbugs.AbstractBugReporter.reportBug(AbstractBugReporter.java:179)
 ...
     [java]       At edu.umd.cs.findbugs.FindBugs.runMain(FindBugs.java:395)
     [java]       At edu.umd.cs.findbugs.FindBugs2.main(FindBugs2.java:1231)
     [java]   Exception analyzing com.example.in.whatever.package.ClassNN using detector edu.umd.cs.findbugs.detect.RuntimeExceptionCapture
     [java]     java.lang.ArrayIndexOutOfBoundsException: 5
     [java]       At edu.umd.cs.findbugs.BugCounts.addError(PackageStats.java:50)
     [java]       At edu.umd.cs.findbugs.PackageStats.addError(PackageStats.java:240)
     [java]       At edu.umd.cs.findbugs.ProjectStats.addBug(ProjectStats.java:272)
     [java]       At edu.umd.cs.findbugs.SortedBugCollection.add(SortedBugCollection.java:960)
     [java]       At edu.umd.cs.findbugs.SortedBugCollection.add(SortedBugCollection.java:227)
     [java]       At edu.umd.cs.findbugs.BugCollectionBugReporter.doReportBug(BugCollectionBugReporter.java:100)
     [java]       At edu.umd.cs.findbugs.AbstractBugReporter.reportBug(AbstractBugReporter.java:179)
     [java]       At edu.umd.cs.findbugs.DelegatingBugReporter.reportBug(DelegatingBugReporter.java:69)
     [java]       At edu.umd.cs.findbugs.FilterBugReporter.reportBug(FilterBugReporter.java:49)
     [java]       At edu.umd.cs.findbugs.BugAccumulator.reportBug(BugAccumulator.java:199)
     [java]       At edu.umd.cs.findbugs.BugAccumulator.reportAccumulatedBugs(BugAccumulator.java:180)
     [java]       At edu.umd.cs.findbugs.detect.RuntimeExceptionCapture.visitJavaClass(RuntimeExceptionCapture.java:118)
     [java]       At org.apache.bcel.classfile.JavaClass.accept(JavaClass.java:213)
     [java]       At edu.umd.cs.findbugs.BytecodeScanningDetector.visitClassContext(BytecodeScanningDetector.java:38)
     [java]       At edu.umd.cs.findbugs.DetectorToDetector2Adapter.visitClass(DetectorToDetector2Adapter.java:76)
     [java]       At edu.umd.cs.findbugs.FindBugs2.lambda$analyzeApplication$1(FindBugs2.java:1108)
     [java]       At java.util.concurrent.FutureTask.run(FutureTask.java:266)
     [java]       At edu.umd.cs.findbugs.CurrentThreadExecutorService.execute(CurrentThreadExecutorService.java:86)
     [java]       At java.util.concurrent.AbstractExecutorService.invokeAll(AbstractExecutorService.java:238)
     [java]       At edu.umd.cs.findbugs.FindBugs2.analyzeApplication(FindBugs2.java:1118)
     [java]       At edu.umd.cs.findbugs.FindBugs2.execute(FindBugs2.java:309)
     [java]       At edu.umd.cs.findbugs.FindBugs.runMain(FindBugs.java:395)
     [java]       At edu.umd.cs.findbugs.FindBugs2.main(FindBugs2.java:1231)
     [java] Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 5
     [java] 	at edu.umd.cs.findbugs.BugCounts.addError(PackageStats.java:50)
     [java] 	at edu.umd.cs.findbugs.PackageStats.addError(PackageStats.java:240)
     [java] 	at edu.umd.cs.findbugs.ProjectStats.addBug(ProjectStats.java:272)
     [java] 	at edu.umd.cs.findbugs.SortedBugCollection.add(SortedBugCollection.java:960)
     [java] 	at edu.umd.cs.findbugs.SortedBugCollection.add(SortedBugCollection.java:227)
     [java] 	at edu.umd.cs.findbugs.BugCollectionBugReporter.doReportBug(BugCollectionBugReporter.java:100)
     [java] 	at edu.umd.cs.findbugs.AbstractBugReporter.reportBug(AbstractBugReporter.java:179)
     [java] 	at edu.umd.cs.findbugs.DelegatingBugReporter.reportBug(DelegatingBugReporter.java:69)
     [java] 	at edu.umd.cs.findbugs.FilterBugReporter.reportBug(FilterBugReporter.java:49)
     [java] 	at edu.umd.cs.findbugs.detect.UnreadFields.report(UnreadFields.java:1063)
     [java] 	at edu.umd.cs.findbugs.DetectorToDetector2Adapter.finishPass(DetectorToDetector2Adapter.java:55)
     [java] 	at edu.umd.cs.findbugs.FindBugs2.analyzeApplication(FindBugs2.java:1154)
     [java] 	at edu.umd.cs.findbugs.FindBugs2.execute(FindBugs2.java:309)
     [java] 	at edu.umd.cs.findbugs.FindBugs.runMain(FindBugs.java:395)
     [java] 	at edu.umd.cs.findbugs.FindBugs2.main(FindBugs2.java:1231)
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for MyProject Parent 3.4.0-SNAPSHOT:
[INFO] 
[INFO] MyProject Parent ................................... SUCCESS [ 15.419 s]
[INFO] MyProject First Module ............................. SUCCESS [06:03 min]
[INFO] MyProject Second Module ............................ SUCCESS [02:13 min]
[INFO] MyProject Third module ............................. SUCCESS [ 52.992 s]
[INFO] MyProject Fourth Module ............................ SUCCESS [ 44.946 s]
[INFO] MyProject Fifth Module ............................. SUCCESS [ 46.276 s]
[INFO] MyProject Sixth Module ............................. FAILURE [ 52.726 s]
[INFO] MyProject Seventh Module ........................... SKIPPED
[INFO] MyProject Eighth Module ............................ SKIPPED
[INFO] MyProject Nineth Module ............................ SKIPPED
[INFO] MyProject Tenth Module ............................. SKIPPED
[INFO] MyProject Eleventh Module .......................... SKIPPED
[INFO] MyProject Twelveth Module .......................... SKIPPED
[INFO] MyProject Thirteenth Module ........................ SKIPPED
[INFO] MyProject Fourteenth Module ........................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  11:50 min
[INFO] Finished at: 2023-01-20T20:08:11+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.spotbugs:spotbugs-maven-plugin:4.7.3.0:spotbugs (default) on project my-project-module6: Execution default of goal com.github.spotbugs:spotbugs-maven-plugin:4.7.3.0:spotbugs failed: Java returned: 1 -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :my-project-module6

Process finished with exit code 1
```

Note that :
- The multiple occurrences of ArrayIndexOutOfBoundsException appear for the 7 modules that got built until the failure
- At each module, the multiple occurrences of ArrayIndexOutOfBoundsException feel like they are for each and every class in that module

**Comment by Ceddaerrix** [comment]
I have solved the problem in my project by setting the _maxHeap_ property as follow:
```
<plugin>
  <groupId>com.github.spotbugs</groupId>
  <artifactId>spotbugs-maven-plugin</artifactId>
  <version>4.7.3.0</version>
	<executions>
		<execution>
			<phase>verify</phase>
			<goals>
				<goal>spotbugs</goal>
			</goals>
		</execution>
	</executions>
	<configuration>
		<failsOnError>false</failsOnError>
		<relaxed>true</relaxed>
		<xmlOutput>true</xmlOutput>
		<effort>more</effort>
                **<maxHeap>1536</maxHeap>**
	</configuration>
</plugin>
```

**Comment by gtoison** [comment]
Can you try with 'relaxed' set to false?
From what I understand the problem is that in relaxed mode all bugs should be reported but some other parts do not expect bugs with priority IGNORED

**Comment by hazendaz** [comment]
Few questions myself on this.  

What jdk is used to build?  Can you use latest maven plugin?  Its been updated numerous times this year.  And finally, try overriding spotbugs with latest snapshot here to see if we didn't already fix that condition.  I've never seen it myself...

**Comment by gtoison** [comment]
This should be fixed by #3000
